src/vm/core.zig

```diff
@@ -482,6 +482,46 @@ pub fn mulOperands(
     return relocatable.fromFelt(op_0_felt.mul(op_1_felt));
 }
 
+/// Attempts to deduce `op1` and `res` for an instruction, given `dst` and `op0`.
+///
+/// # Arguments
+/// - `inst`: The instruction to deduce `op1` and `res` for.
+/// - `dst`: The destination of the instruction.
+/// - `op0`: The first operand of the instruction.
+///
+/// # Returns
+/// - `Tuple`: A tuple containing the deduced `op1` and `res`.
+pub fn deduceOp1(
+    inst: *const instructions.Instruction,
+    dst: ?*const relocatable.MaybeRelocatable,
+    op0: ?*const relocatable.MaybeRelocatable,
+) !std.meta.Tuple(&[_]type{ ?relocatable.MaybeRelocatable, ?relocatable.MaybeRelocatable }) {
+    if (inst.opcode != .AssertEq) {
+        return .{ null, null };
+    }
+
+    switch (inst.res_logic) {
+        .Op1 => if (dst) |dst_val| {
+            return .{ dst_val.*, dst_val.* };
+        },
+        .Add => if (dst != null and op0 != null) {
+            return .{ try dst.?.sub(op0.?.*), dst.?.* };
+        },
+        .Mul => {
+            if (dst == null or op0 == null)
+                return .{ null, null };
+            if (dst.?.isRelocatable() or op0.?.isRelocatable())
+                return .{ null, null };
+            if (op0.?.felt.isZero())
+                return .{ null, null };
+            return .{ relocatable.fromFelt(try dst.?.felt.div(op0.?.felt)), dst.?.* };
+        },
+        else => {},
```

Comment:
I think this could be simplified further to return nulls at one place, but i don't have a better idea at this moment

Reply:
It'a a bit better than what I had before, but yeah. Still not ideal.

The easiest is to do something like this:

```zig
if (dst != null and op0 != null and !dst.?.isRelocatable() and !op0.?.isRelocatable() and !op0.?.felt.isZero())
    return .{ relocatable.fromFelt(try dst.?.felt.div(op0.?.felt)), dst.?.* };
```

But the auto formatter keeps putting this on one line, making it hard to read and hard to reason about properly.

Reply:
That should work if you manually introduce newlines after the 'and's, and I think you might have to put a comma after `dst.?.*`

Reply:
oooh the `and` has to go at the end of the line!

Should I change it?